tensornetwork/matrixproductstates/base_mps.py

```diff
@@ -17,10 +17,13 @@
 from __future__ import print_function
 import numpy as np
 # pylint: disable=line-too-long
-from tensornetwork.network_components import Node, contract, contract_between, BaseNode
+from tensornetwork.network_components import Node, contract_between
 # pylint: disable=line-too-long
-from tensornetwork.network_operations import split_node_qr, split_node_rq, split_node_full_svd, norm, conj, switch_backend
+from tensornetwork.network_operations import split_node_full_svd, conj
+from tensornetwork.backends import backend_factory
 from typing import Any, List, Optional, Text, Type, Union, Dict, Sequence
+from tensornetwork import ncon
```

Comment:
Aw, here is where the cyclic import is happening. You need to do from `tensornetwork.ncon import ncon`

Reply:
So what happened is `from tensornetwork import ncon` actually loads the `ncon` function from `tensornetwork/__init__.py`

Reply:
thanks!!! I can finally go to sleep :)